services/code.service.js

```
@@ -611,6 +611,127 @@ const _installDependencies = async (path) => {
+            if (!isRejected) {
+                isRejected = true
+                reject(err)
+            }
+        })
+    })
+}
+
+const _runVitestTests = async (path) => {
+    logger.log('Starting Vitest server in path: ', path)
+
+    return new Promise((resolve, reject) => {
+        const vitestServer = spawn('yarn', ['test:json'], {
+            cwd: path,
+            detached: true,
+        })
+
+        let isRejected = false
+        let stdout = ''
+        let stderr = ''
+
+        vitestServer.stdout.on('data', (data) => {
+            const output = data.toString()
+            stdout += output
+
+            // Attempt to parse JSON output and resolve the promise
+            try {
+                const jsonStartIndex = stdout.indexOf('{')
+                const jsonEndIndex = stdout.lastIndexOf('}')
+                if (jsonStartIndex !== -1 && jsonEndIndex !== -1) {
+                    const jsonString = stdout.slice(jsonStartIndex, jsonEndIndex + 1)
+                    const parsedJSON = JSON.parse(jsonString)
+
+                    const result = {
+                        success: [],
+                        failed: [],
+                    }
+
+                    for (const testResult of parsedJSON?.testResults || []) {
+                        for (const assertion of testResult?.assertionResults || []) {
+                            if (assertion?.status === 'passed') {
+                                result?.success?.push(assertion?.fullName)
+                            } else if (assertion?.status === 'failed') {
+                                result?.failed?.push(assertion?.fullName)
+                            }
+                        }
+                    }
+
+                    resolve(result)
+                    vitestServer.kill() // Terminate the process once JSON is parsed
+                }
+            } catch (err) {
+                // Ignore JSON parsing errors until complete JSON is received
```

Lets atleast log it , because an empty catch will not make sense.

---

resolved by https://github.com/kalviumcommunity/compilerd/pull/167/commits/dc7d35b094dc701d0f77fb362502877541e18044